@xen-orchestra/web-core/docs/guidelines/component-example.md

```diff
@@ -0,0 +1,242 @@
+# Component example
+
+Here is a fake `UiComponent.vue` component that uses most of the guidelines documented:
+
+```vue
+<!-- v4 -->
+<template>
+  <!-- Root class name matches the component's name -->
+  <div class="ui-component" :class="classNames">
+    <!-- Default slot-->
+    <slot />
+
+    <!-- Child component used only in this component-->
+    <!-- Note the shorthand version for "icon" prop binding -->
+    <SquareIcon :icon accent="current" class="icon" />
+
+    <!-- Render the element only if the slot is used -->
+    <div v-if="slots.info">
+      <slot name="info" />
+    </div>
+
+    <!-- Slot + prop pattern -->
+    <slot name="message">
+      <p>{{ message }}</p>
+    </slot>
```

Comment:
```suggestion
    <slot v-if="slots.message" name="message">
      <p>{{ message }}</p>
    </slot>
```

Reply:
In this case, it doesn’t make sense to add a `v-if` directly on the slot. This check is used when the slot is a child of an HTML element, and we don't want to render an empty element if the slot is not used.
However, the case is special here because I added an unnecessary `<p>` element. I will remove it.